wgengine/magicsock/magicsock.go

```diff
@@ -517,7 +517,7 @@ func appendDests(dsts []*net.UDPAddr, as *AddrSet, b []byte) (_ []*net.UDPAddr,
 			return dsts, roamAddr
 		}
 	}
-	for i := len(as.addrs) - 1; i >= 0; i-- {
+	for i := range as.addrs {
 		addr := &as.addrs[i]
 		if spray || as.curAddr == -1 || as.curAddr == i {
```

Comment:
Until @crawshaw added the time stuff in this func, it was pretty nicely testable (but untested).

I can add tests to it in a follow-up CL, unless you want to do it in this CL. I think you could just in the now parameter to it, and have the caller do the time.Now() (and the tests).

(Commenting in random place because GitHub)

Reply:
A `timeNow` method should do the trick.

Reply:
Getting this submitted now so we can keep banging on new Testflight builds, but yes, I take your point that time should be plumbed through this better.